test/parallel/test-child-process-spawn-name.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+
+// This test spawns itself with an argument to indicate when it is a child to
+// easily and portably print the process title
+if (process.argv.length == 3) {
+  console.log(process.title);
+  process.exit();
+}
+
+const noCustomName = cp.spawn('node',
```

Comment:
Just kidding, wasn't thinking. I can compare against `execPath` in the check too. 

Fixed.